EIPS/eip-5075.md

```diff
@@ -0,0 +1,293 @@
+                    //transfers out
+                    payable(to).transfer(amount);
+                    return(true);
+                }
+                //if the token is a ERC20 token
+            } else {
+                //used to get around solidity 0.8 reverts
+                unchecked {
+                if (block.timestamp != info.timestamp)
+                    {
+                    limitUnlocked = uint224(
+                        (IERC20(token).balanceOf(address(this)) * rateLimit) /
+                        (timeWindow / (block.timestamp - info.timestamp)) /
+                        10000
+                    );
+                    }
+                    if (info.ratedFlow <= limitUnlocked || timeWindow <= block.timestamp - info.timestamp) {
+                        info.ratedFlow = 0;
+                    } else {
+                        info.ratedFlow -= uint224(limitUnlocked);
+                    }
+                }
+                //increases the tracked ratedFlow for the current time window by the amount sent out
+                info.ratedFlow += uint224(amount);
+                unchecked {
+                    //revert if the outflow exceeds rate limit
+                    require(
+                        info.ratedFlow <=
+                        (rateLimit *
+                        IERC20(token).balanceOf(address(this))) /
+                        10000
+                    );
+                    //sets the current time as the last transfer for the token
+                    info.timestamp = uint32(block.timestamp);
+                    //transfers out
+                    return(IERC20(token).transfer(to, amount));
+                }
+            }
+        }
+    }
+}
+```
+**Note**
+
+The actual limit would be based on the previous 1h outflows and should that be 0 the limit approaches 2*rateLimit for that 1h slot, though to be noted does not apply for following slots so rather than 2x, net outflow limit for any time t can be seen as
+
+rateLimit available at time 0 + timeWindow  * rateLimit
+
+Not simply timeWindow * rateLimit, where 2 * rateLimit for cases where the start outflow (at time 0 for the window) would be 0 for the previous 1h (no activity last 1h), as the outflow at time 0 increases the excess possible decreases.
+
+So let's say that you have a timeWindow at 100m and a rateLimit at 100 bips (lets say theres 1000 tokens and so 100 tokens per 100m)
+```
+Time (m)
+Amount out (up to limit)
+000 005 010 020 040 060 080 099 100 105 110 120...
+100 005 010 010 020 020 020 014 001 005 005 010...
+Delta 1h
+100+005+005+010+020+020+020+019+001 (200 last 1h possible at 0-100m passed)
+....005+005+010+020+020+020+019+001+005 (105 last 1h possible at 05-105m passed)
+
+Amount out last 1h window
+100 105 115 125 145 165 185 199 200 105 105 110 ...
+```
+
+So realistically ends up as the excess to the rateLimit possible being the free rate available at time 0 for the window.
+
+**Note**
```

Comment:
There is another `Note` subsection immediately above this, so I'm guessing this was an accidental double inclusion?
```suggestion
```